python/pyarrow/types.pxi

```diff
@@ -967,7 +983,25 @@ cpdef DataType decimal(int precision, int scale=0):
     decimal_type : DecimalType
     """
     cdef shared_ptr[CDataType] decimal_type
-    decimal_type.reset(new CDecimalType(precision, scale))
+    decimal_type.reset(new CDecimalType(byte_width, precision, scale))
+    return pyarrow_wrap_data_type(decimal_type)
```

Comment:
`arrow::DecimalType` is the base type now -- I don't think we should allow users to create instances of it

Reply:
Should I simply remove this function then?

Reply:
Yes, and the `CDecimalType` class in libarrow.pxd

Reply:
Actually, that's OK to leave